src/EnergyPlus/UnitVentilator.cc

```diff
@@ -804,469 +650,364 @@ namespace UnitVentilator {
             }
 
             // Get Coil information
-            if (state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption == BothOption ||
-                state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption == HeatingOption) {
-                // Heating coil information:
-                // A14, \field Heating Coil Object Type
-                //      \type choice
-                //      \key Coil:Heating:Water
-                //      \key Coil:Heating:Electric
-                //      \key Coil:Heating:Fuel
-                //      \key Coil:Heating:Steam
-                // A15, \field Heating Coil Name
-                //      \type object-list
-                //      \object-list HeatingCoilName
+            if (unitVent.CoilOption == CoilOption::Both || unitVent.CoilOption == CoilOption::Heating) {
                 if ((!lAlphaBlanks(16))) {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilPresent = true;
+                    unitVent.HCoilPresent = true;
                     errFlag = false;
 
                     cHeatingCoilType = Alphas(15);
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilTypeCh = cHeatingCoilType;
+                    unitVent.HCoilTypeCh = cHeatingCoilType;
                     {
-                        auto const SELECT_CASE_var(cHeatingCoilType);
-                        if (SELECT_CASE_var == "COIL:HEATING:WATER") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilType = Heating_WaterCoilType;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HeatingCoilType = DataPlant::PlantEquipmentType::CoilWaterSimpleHeating;
-                        } else if (SELECT_CASE_var == "COIL:HEATING:STEAM") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilType = Heating_SteamCoilType;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HeatingCoilType = DataPlant::PlantEquipmentType::CoilSteamAirHeating;
-                        } else if (SELECT_CASE_var == "COIL:HEATING:ELECTRIC") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilType = Heating_ElectricCoilType;
-                        } else if (SELECT_CASE_var == "COIL:HEATING:FUEL") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).HCoilType = Heating_GasCoilType;
+                        auto const heatingCoilType(cHeatingCoilType);
```

Comment:
`getEnumerationValue`. And then either a `std::array` map from `HCoilType` to `HeatingCoilType` or get rid of one of them.